### PR TITLE
Storybook: Fix asset paths for hosted storybooks

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
Published storybook currently contains 1000s of images that are never used. From time to time this causes the drone publishing storybook step to fail. This PR limits these images to only the available icons and the storybook theme assets.

The font face rules in our sass use relative paths that look up a level for the fonts folder. This fails in a published storybook as it looks outside the "latest" or "canary" folder for a fonts directory which doesn't exist. This PR also addresses this by introducing a sass var which is then overridden in sass theme files specific for storybook.

**Special notes for your reviewer**:

 @dsotirakis these changes along with #56309 should solve the issues we discussed today.

### CODE REVIEW

File: packages/grafana-ui/tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
   // override for storybook which uses ts-node to compile main.ts / preview.ts files.
   "ts-node": {
     "compilerOptions": {
-      "isolatedModules": false
+      "isolatedModules": false,
+      "module": "commonjs"
```
Comment (jackw) [inline packages/grafana-ui/tsconfig.json:16]: This is required to use `import {} from ...` in `main.ts`.

File: public/sass/components/_dashboard_grid.scss [hunk 2]
```diff
@@ -1,5 +1,5 @@
-@import 'react-grid-layout/css/styles.css';
-@import 'react-resizable/css/styles.css';
+@import 'react-grid-layout/css/styles';
```
Comment (jackw) [inline public/sass/components/_dashboard_grid.scss:1]: Removing the `.css` from these imports allows sass to treat them as sass imports and bundle the css into the built sass. Previously these were being compiled to [css imports](https://sass-lang.com/documentation/at-rules/import#plain-css-imports) and caused storybook to fail to compile the sass as it couldn't resolve the css files.

**Review by dsotirakis — COMMENTED** [review]
Thanks for doing that @jackw 🙇 
I'll let someone else review this PR since it's too frontend-y for me!!!

**Review by academo — APPROVED** [review]

**Review by mckn — APPROVED** [review]
LGTM! Great work with this 👍🏻

**Pull request merged**